src/SBInterpolatedImage.cpp

```diff
@@ -900,6 +782,198 @@ namespace galsim {
         return result;
     }
 
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    // SBInterpolatedKImage methods
+
+    template <typename T>
+    SBInterpolatedKImage::SBInterpolatedKImage(
+        const BaseImage<T>& realImage,
+        const BaseImage<T>& imagImage,
+        double dk, double stepk,
+        boost::shared_ptr<Interpolant> kInterp,
+        const GSParamsPtr& gsparams) :
+        SBInterpolated(new SBInterpolatedKImageImpl(realImage, imagImage, dk, stepk,
+                boost::shared_ptr<Interpolant2d>(new InterpolantXY(kInterp)), gsparams)
+        ) {}
+
+    template <typename T>
+    SBInterpolatedKImage::SBInterpolatedKImage(
+        const BaseImage<T>& realImage,
+        const BaseImage<T>& imagImage,
+        double dk, double stepk,
+        boost::shared_ptr<Interpolant2d> kInterp,
+        const GSParamsPtr& gsparams) :
+        SBInterpolated(new SBInterpolatedKImageImpl(realImage, imagImage, dk, stepk,
+                                                    kInterp, gsparams)
+        ) {}
+
+    SBInterpolatedKImage::SBInterpolatedKImage(const SBInterpolatedKImage& rhs)
+        : SBInterpolated(rhs) {}
+
+    SBInterpolatedKImage::~SBInterpolatedKImage() {}
+
+    ConstImageView<double> SBInterpolatedKImage::getRealKImage() const
+    {
+        assert(dynamic_cast<const SBInterpolatedKImageImpl*>(_pimpl.get()));
+        return static_cast<const SBInterpolatedKImageImpl&>(*_pimpl).getRealKImage();
+    }
+
+    ConstImageView<double> SBInterpolatedKImage::getImagKImage() const
+    {
+        assert(dynamic_cast<const SBInterpolatedKImageImpl*>(_pimpl.get()));
+        return static_cast<const SBInterpolatedKImageImpl&>(*_pimpl).getImagKImage();
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    // SBInterpolatedKImageImpl methods
+
+    template <typename T>
+    SBInterpolatedKImage::SBInterpolatedKImageImpl::SBInterpolatedKImageImpl(
+        const BaseImage<T>& realKImage, const BaseImage<T>& imagKImage,
+        double dk, double stepk, boost::shared_ptr<Interpolant2d> kInterp,
+        const GSParamsPtr& gsparams) :
+        SBInterpolatedImpl(kInterp, stepk, 0., gsparams), //fill in maxk below
+        _dk(dk)
+    {
+        dbg<<"stepk = "<<stepk<<std::endl;
+        dbg<<"kimage bounds = "<<realKImage.getBounds()<<std::endl;
+        assert(_kInterp.get());
+
+        Ninitial = std::max(realKImage.getXMax()-realKImage.getXMin()+1,
+                            realKImage.getYMax()-realKImage.getYMin()+1);
+        dbg<<"Ninitial = "<<Ninitial<<std::endl;
+        Nk = goodFFTSize(int(Ninitial));
+        dbg<<"Nk = "<<Nk<<std::endl;
+
+        _ktab = boost::shared_ptr<KTable>(new KTable(Nk, _dk));
```

Comment:
Probably worth adding a comment in here somewhere about the difference between dk and stepk, since they are both nominally the same thing.  So it's worth clarifying for future developers who may need to understand this code.

dk is the scale of the input image, and the KTable is constructed with the same scale.  stepk on the other hand is the maximum scale that a *rendered* Fourier image will have.  So stepk is only used in the calculations done by drawImage to figure out how to render an image, not in any of the stuff here.